csharp/src/Drivers/Apache/Hive2/HiveServer2TlsImpl.cs

```diff
@@ -0,0 +1,106 @@
+﻿/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using System.Net.Http;
+
+namespace Apache.Arrow.Adbc.Drivers.Apache.Hive2
+{
+    class TlsProperties
+    {
+        public bool IsSslEnabled { get; set; }
+        public bool EnableServerCertificateValidation { get; set; }
+        public bool AllowHostnameMismatch { get; set; }
+        public bool AllowSelfSigned { get; set; }
+        public string? TrustedCertificatePath { get; set; }
+    }
+    static class HiveServer2TlsImpl
+    {
+        static internal TlsProperties GetTlsOptions(IReadOnlyDictionary<string, string> properties)
+        {
+            TlsProperties tlsProperties = new TlsProperties();
+            properties.TryGetValue(TlsOptions.IsSslEnabled, out string? isSslEnabled);
+            properties.TryGetValue(AdbcOptions.Uri, out string? uri);
+            if (!string.IsNullOrWhiteSpace(uri))
```

Comment:
This generally assumes that the `out` parameter has a specific well-defined value even when the `Try` returns false. I think it's a better practice to structure the condition as `if (properties.TryGetValue(AdbcOptions.Uri, out string? uri) && !string.IsNullOrWhiteSpace(uri))`. (Same applies to all `TryGetValue` calls in this method.)